wallet/core/src/tx/generator/generator.rs

```diff
@@ -302,6 +327,34 @@ impl Generator {
             return Err(Error::GeneratorChangeAddressNetworkTypeMismatch);
         }
 
+        let standard_change_output_mass =
+            mass_calculator.calc_mass_for_output(&TransactionOutput::new(0, pay_to_address_script(&change_address)));
+        let signature_mass_per_input = mass_calculator.calc_signature_mass(minimum_signatures);
+        let final_transaction_outputs_compute_mass = mass_calculator.calc_mass_for_outputs(&final_transaction_outputs);
+        let final_transaction_payload = final_transaction_payload.unwrap_or_default();
+        let final_transaction_payload_mass = mass_calculator.calc_mass_for_payload(final_transaction_payload.len());
+        let final_transaction_outputs_harmonic =
+            mass_calculator.calc_storage_mass_output_harmonic(&final_transaction_outputs).ok_or(Error::MassCalculationError)?;
+
+        // reject transactions where the payload and outputs are more than 2/3rds of the maximum tx mass
+        let final_transaction = final_transaction_amount.map(|amount| FinalTransaction {
+            value_no_fees: amount,
+            value_with_priority_fee: amount + final_transaction_priority_fee.additional(),
+        });
+
+        let mass_sanity_check = standard_change_output_mass + final_transaction_outputs_compute_mass + final_transaction_payload_mass;
+        if mass_sanity_check > MAXIMUM_STANDARD_TRANSACTION_MASS / 5 * 4 {
+            return Err(Error::GeneratorTransactionOutputsAreTooHeavy { mass: mass_sanity_check, kind: "compute mass" });
+        }
+
+        if let Some(final_transaction) = &final_transaction {
+            let perfect_output_storage_mass =
```

Comment:
lol.  Under "perfect" I sort of meant worse-case scenario...  Let me comment this out but keep it in the comment form with a TODO for further review.  This is where it would be perhaps desirable to do some pre-analysis.